src/Plugin/VoteUpDownWidgetInterface.php

```diff
@@ -24,40 +25,11 @@ public function getWidgetId();
   public function getWidgetTemplate();
 
   /**
-   * Alters template variables on render
-   *
-   * 'id' of the widget for the current instance
-   * @param $widget_template
-   * Array containing all info for a plugin instance
-   * @param $variables
-   *
-   * @return mixed
-   */
-  public function alterTemplateVars($widget_template, &$variables);
```

Comment:
I see you tried to simplify vote values calculations by doing it once via 2 queries and then some logic to get the needed values. But that also means it is 2 queries per widget always. E.g. using the plain widget on comments, in a long thread of comments scenario means 2*number_of_comments queries, which is the double number of queries.

Because of this, I still think using an "alter" interaction point as was defined here is a better idea. Please at least leave the interaction point there to be used, and define it as empty on the widget plugin base class, and better yet if you can change this to do it as it was originally: different queries depending on the widget case.